rpc-rs/src/provider.rs

```diff
@@ -652,15 +677,18 @@ impl HostBridge {
             .map_err(|e| RpcError::Nats(e.to_string()))?;
         self.add_subscription(sub.clone()).await;
         let (this, provider) = (self.clone(), provider.clone());
-        tokio::spawn(async move {
-            while let Some(msg) = sub.next().await {
-                if let Some(ld) = &this.parse_msg::<LinkDefinition>(&msg, "link.del") {
-                    this.delete_link(&ld.actor_id).await;
-                    // notify provider that link is deleted
-                    provider.delete_link(&ld.actor_id).await;
+        tokio::spawn(
+            async move {
+                while let Some(msg) = sub.next().await {
+                    if let Some(ld) = &this.parse_msg::<LinkDefinition>(&msg, "link.del") {
+                        this.delete_link(&ld.actor_id).await;
+                        // notify provider that link is deleted
+                        provider.delete_link(&ld.actor_id).await;
+                    }
                 }
             }
-        });
+            .instrument(tracing::trace_span!("subscribe_link_del", topic = %link_del_topic)),
```

Comment:
span should be created inside while loop

Reply:
Same thing here as the link put stuff